bitnami/rabbitmq/values.yaml

```diff
@@ -146,11 +146,14 @@ auth:
   ## ref: https://github.com/bitnami/containers/tree/main/bitnami/rabbitmq#environment-variables
   ##
   securePassword: true
-  ## @param auth.existingPasswordSecret Existing secret with RabbitMQ credentials (must contain a value for `rabbitmq-password` key)
-  ## e.g:
-  ## existingPasswordSecret: name-of-existing-secret
+  ## @param auth.existingSecret The name of an existing secret with RabbitMQ credentials
+  ## NOTE: When it's set, the previous `auth.password` parameter is ignored
+  ##
+  existingSecret: ""
+  ## @param auth.existingSecretPasswordKey Password key to be retrieved from existing secret
+  ## NOTE: ignored unless `auth.existingSecret` parameter is set
```

Comment:
Introducing backwards incompatibilities comes with a high cost (specially in terms of UX). Therefore, we should avoid it unless it's strictly necessary.

Here you have a difference alternative so we can keep this change as a minor:

- Mark current `auth.existingPasswordSecret` as deprecated but keeping it:

```yaml
  ## @param auth.existingPasswordSecret Existing secret with RabbitMQ credentials (must contain a value for `rabbitmq-password` key)
  ## DEPRECATED. Use auth.existingSecret instead
  ##
  existingPasswordSecret: ""
  ## @param auth.existingSecret The name of an existing secret with RabbitMQ credentials
  ## NOTE: When it's set, the previous `auth.password` parameter is ignored
  ##
  existingSecret: ""
  ## @param auth.existingSecretPasswordKey Password key to be retrieved from existing secret
  ## NOTE: ignored unless `auth.existingSecret` parameter is set 
  ##
  existingSecretPasswordKey: ""
```

- Then, on `_helpers.tpl` (and other potential templates where the secret name is required) use `coalesce` so you give priority to the new parameter over the deprecated one. For instance:

```yaml
{{/*
Get RabbitMQ password secret name.
*/}}
{{- define "rabbitmq.secretName" -}}
{{- $existingSecret := coalesce .Values.auth.existingSecret .Values.auth.existingPasswordSecret -}}
{{- print (default (include "common.names.fullname" .) (tpl $existingSecret .)) -}}
{{- end -}}
```

Reply:
good suggestion. I found it hard to grasp how to make it backward compatible

Reply:
this would still mean that we need to implement a fallback for the key if the secret name is provided (rabbitmq-password)